mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
```

Comment:
I would say the problem here is the name of the variable ```isDefaultPortForProtocol```. It should be ```isDefaultPort``` and that would make things more clear.

So normalizePort doesn't care about the protocol it just removes the :80 or :443 if they are target ports.

I'm mapping the corresponding protocol in a different place. See in mixed-content-test-case.js:

```
var insecurePort = normalizePort(parseInt("{{ports[http][0]}}"));
  var securePort = normalizePort(parseInt("{{ports[https][0]}}"));

```